packages/@aws-cdk/aws-lambda-python/lib/layer.ts

```diff
@@ -0,0 +1,74 @@
+import * as path from 'path';
+import * as lambda from '@aws-cdk/aws-lambda';
+import * as cdk from '@aws-cdk/core';
+import { bundleLayer } from './bundling';
+
+/**
+ * Properties for PythonDependenciesLayer
+ */
+export interface PythonLayerVersionProps {
```

Comment:
Should this extend LayerVersionProps to stay in sync?

Reply:
I would like to, but LayerVersionProps requires `code` and we're supposed to be responsible for providing that. I'm not aware of an interface that might serve as a good base like `PythonFunctionProps`'s `lambda.FunctionOptions`. I suppose that we could look at implementing something like that in `@aws-cdk/aws-lambda`.

Reply:
I've created `lambda.LayerVersionOptions` in `@aws-cdk/aws-lambda` that I hope will do the trick. I used `lambda.FunctionOptions` as a model, so I separated the runtime options from the rest. But, as a result, I've had to define my own `compatibleRuntimes` in `LayerVersionProps`.

Reply:
I think this is probably a step in the right direction. The duplication of `compatibleRuntimes` feels strange though. I also wonder if a BaseProps interface is in order. I haven't yet wrapped my head around Options vs Props.

I haven't had the time yet to fully digest the interface patterns yet, but it seems like maybe restapi.ts and lambda-api.ts might be a good reference for what these interfaces should look like. I'll dig deeper into it on Monday. I noticed @eladb wrote most of that code so maybe he can supply some guidance.